### PR TITLE
fix(action): replace set-output with environment variable

### LINKED ISSUE #53: GitHub set-output deprecation warnings
```
Run WyriHaximus/github-action-next-semvers@v1
Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

Already updated in `WyriHaximus/github-action-get-previous-tag` that we also use. Much appreciated if you get the chance to sort this one too 🙌

### PR DESCRIPTION
**Pull request by dargmuesli**
Resolves #53.

Please try it though / use it as a starting point for an improved implementation.

### CODE REVIEW

File: src/Next.php [hunk 1]
```diff
@@ -55,26 +55,26 @@ public static function run(string $versionString, bool $strict): string
...
+        $output  = '"major=' . $version->incrementMajor() . '" >> $GITHUB_OUTPUT' . PHP_EOL;
+        $output .= '"minor=' . $version->incrementMinor() . '" >> $GITHUB_OUTPUT' . PHP_EOL;
 
         ///////////////////////////////////////////////////////////////////////////////////////////////
         // v prefixed versions
         ///////////////////////////////////////////////////////////////////////////////////////////////
-        $output .= '::set-output name=v_major::v' . $version->incrementMajor() . PHP_EOL;
-        $output .= '::set-output name=v_minor::v' . $version->incrementMinor() . PHP_EOL;
+        $output .= '"v_major=v' . $version->incrementMajor() . '" >> $GITHUB_OUTPUT' . PHP_EOL;
+        $output .= '"v_minor=v' . $version->incrementMinor() . '" >> $GITHUB_OUTPUT' . PHP_EOL;
```
Comment (WyriHaximus) [inline src/Next.php:65]: Change all of them in this file to this format, and then in `next.php` in the root of this repository replace the `echo` with a file_put_contents as in: https://github.com/WyriHaximus/github-action-supported-php-versions/blob/940773a095e0d019bca8d75262755dcfbb3092f1/versions.php#L27

```suggestion
        $output .= '"v_minor=v' . $version->incrementMinor() . "\"\n";
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -55,11 +55,11 @@ jobs:
           if [ "${{ github.event_name }}" == "pull_request" ] ; then
             ref=$(php -r "echo str_replace('/', '-SLASH-', '${{ github.event.pull_request.head.ref }}');")
             echo "$ref"
-            printf "::set-output name=ref::%s" $ref
+            printf "ref=%s >> $GITHUB_OUTPUT" $ref
```
Comment (WyriHaximus) [inline .github/workflows/ci.yml:58]:
```suggestion
            printf "ref=%s" $ref >> $GITHUB_OUTPUT 
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -55,11 +55,11 @@ jobs:
...
             exit 0
           fi
           echo "${GITHUB_REF##*/}"
-          echo "::set-output name=ref::${GITHUB_REF##*/}"
+          echo "ref=${GITHUB_REF##*/} >> $GITHUB_OUTPUT"
```
Comment (WyriHaximus) [inline .github/workflows/ci.yml:62]:
```suggestion
          echo "ref=${GITHUB_REF##*/}" >> $GITHUB_OUTPUT
```

**Review by WyriHaximus — CHANGES_REQUESTED** [review]
Thanks for filing these changes. I've requested a few changes. As per your comment https://github.com/WyriHaximus/github-action-next-semvers/issues/53#issuecomment-1350309082 PHP isn't you language but that is fine I gladly guide you through it or take this as a base. There where also a few things off for the action files. 

You might know or not know this but the `>>` in shell takes the output (from STDOUT) and writes it to something on the filesystem. In this case it is the file defined in `$GITHUB_OUTPUT` so the way you are currently implementing this doesn't do that because it is part of the output.

**Comment by dargmuesli** [comment]
Thank you for the quick feedback! I hope I understood your suggestions correctly? :)

File: src/Next.php [hunk 4]
```diff
@@ -55,26 +55,26 @@ public static function run(string $versionString, bool $strict): string
...
             // v prefixed versions
-            $output .= '::set-output name=v_patch::v' . $version . PHP_EOL;
+            $output .= '"v_patch=v' . $version . "\"\n";
         } else {
             // increment major/minor/patch version
-            $output .= '::set-output name=patch::' . $version->incrementPatch() . PHP_EOL;
+            $output .= '"patch=' . $version->incrementPatch() . "\"\n";
             // v prefixed versions
-            $output .= '::set-output name=v_patch::v' . $version->incrementPatch() . PHP_EOL;
+            $output .= '"v_patch=v' . $version->incrementPatch();
```
[the remark this answers is not in the document]
Comment (WyriHaximus) [reply src/Next.php:77]: Needs to be on all of them, something else might write to it as well after this action and that will mess it potentially up.

**Review by WyriHaximus — CHANGES_REQUESTED** [review]

File: src/Next.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (dargmuesli) [reply src/Next.php]: done

**Comment by WyriHaximus** [comment]
> Thank you for the quick feedback! I hope I understood your suggestions correctly? :)

Yes, you did :D. It mostly looks good aside from that one comment and the change that needs to be made on this line: https://github.com/WyriHaximus/github-action-next-semvers/blob/master/next.php#L14

**Comment by dargmuesli** [comment]
What else needs to be done there? :sweat_smile: I thought I did the change as intended :see_no_evil:

**Comment by WyriHaximus** [comment]
> What else needs to be done there? 😅 I thought I did the change as intended 🙈

Well you did, except that you still need to change https://github.com/WyriHaximus/github-action-next-semvers/blob/master/next.php#L14 to 
```php
file_put_contents(getenv('GITHUB_OUTPUT'), Next::run($versionString, \getenv('INPUT_STRICT') === 'false' ? false : true), FILE_APPEND);
```

**Comment by dargmuesli** [comment]
I think we have a misunderstanding. That line is already be changed as far as I get.

File: next.php [hunk 5]
```diff
@@ -11,7 +11,7 @@
         return 1;
     }
 
-    echo Next::run($versionString, \getenv('INPUT_STRICT') === 'false' ? false : true);
+    file_put_contents(getenv('GITHUB_OUTPUT'), Next::run($versionString, \getenv('INPUT_STRICT') === 'false' ? false : true), FILE_APPEND);
```
Comment (dargmuesli) [inline next.php:14]: Are we talking about this change?

Comment (WyriHaximus) [reply next.php:14]: Yes, wait did I completely miss that? I'm so sorry, I had a rough couple of weeks and I was expecting the file under the directories in the tree on the left :facepalm:

**Review by WyriHaximus — APPROVED** [review]
Yeah completely missed that, I'll merge and get this released later today. Thank you for all the work on it :+1: .

**Pull request merged**